addons/binding/org.openhab.binding.exec/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="exec"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+        xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+    <thing-type id="command">
+        <label>Command</label>
+        <description>The Command Thing encapsulates a shell command to be executed</description>
+
+        <channels>
+            <channel id="output" typeId="output"/>
+            <channel id="exit" typeId="exit"/>
+            <channel id="running" typeId="running"/>
+            <channel id="execute" typeId="execute"/>
+            <channel id="lastexecution" typeId="lastexecution" />
+        </channels>
+        
+        <config-description>
+            <parameter name="command" type="text" required="true">
+                <label>Command</label>
+                <description>The command to execute</description>
+            </parameter>
+            <parameter name="transform" type="text" required="false">
+                <label>Transform</label>
+                <description>The transformation to apply on the execution result</description>
+            </parameter>
+            <parameter name="interval" type="integer"  required="false">
+                <label>Interval</label>
+                <description>Interval, in seconds, the command will be repeatedly executed</description>
+            </parameter>
+            <parameter name="timeout" type="integer"  required="false">
+                <label>Timeout</label>
+                <description>Time out, in seconds, the execution of the command will time out</description>
+            </parameter>        
+        </config-description>
+        
+    </thing-type>
+
+    <channel-type id="output">
+        <item-type>String</item-type>        
+        <label>Output</label>
+        <description>Output of the last execution of the command</description>
+        <state readOnly="true"></state>  
+    </channel-type>
+        <channel-type id="exit">
+        <item-type>Number</item-type>
+        <label>Exit Value</label>
+        <description>The exit value of the last execution of the command</description>
+        <state readOnly="true"></state>  
+    </channel-type>
+     <channel-type id="running">
+        <item-type>Switch</item-type>
+        <label>Running</label>
+        <description>Indicator whether the command is currently being executed</description>
+        <state readOnly="true"></state>  
+    </channel-type>
+     <channel-type id="execute">
```

Comment:
I'd remove this channel and combine it with "running" into a "run" channel. You can send ON to execute the command and the current state tells you whether it is running or not.